test/extensions/clusters/redis/redis_cluster_integration_test.cc

```diff
@@ -136,13 +136,15 @@ class RedisClusterIntegrationTest : public testing::TestWithParam<Network::Addre
       }
     });
 
-    BaseIntegrationTest::initialize();
+    on_server_ready_function_ = [this](Envoy::IntegrationTestServer& test_server) {
+      mock_rng_ = dynamic_cast<Runtime::MockRandomGenerator*>(&(test_server.server().random()));
+      // Abort now if we cannot downcast the server's random number generator pointer.
+      ASSERT_TRUE(mock_rng_ != nullptr);
+      // Ensure that fake_upstreams_[0] is the load balancer's host of choice by default.
+      ON_CALL(*mock_rng_, random()).WillByDefault(Return(random_index_));
```

Comment:
qq: f we are trying to hard code it to a single uptream, why don't we just have a single upstream in the test?

Reply:
we're forcing the service discovery(cluster slots) requests to a single upstream, but all the other Redis requests are sent to multiple upstreams in the tests according to redis cluster protocol.

Reply:
OK goit it, makes sense.